test/python/transpiler/test_calibrationbuilder.py

```diff
@@ -47,140 +47,84 @@ class TestCalibrationBuilder(QiskitTestCase):
     __angle = np.pi / 2
     __granularity = 16
 
-    def setUp(self):
-        super().setUp()
-        # This backend provides CX(0,1) with native ECR direction.
-        self.backend = FakeHanoi()
-        self.inst_map = self.backend.defaults().instruction_schedule_map
+    @staticmethod
+    def get_cr_play(cr_schedule, name):
+        """A helper function to filter CR pulses."""
 
-        # Get positive CR tone
-        def _get_cr(time_inst, name):
+        def _filter_func(time_inst):
             return isinstance(time_inst[1], Play) and time_inst[1].pulse.name.startswith(name)
 
-        cx_sched = self.inst_map.get("cx", (0, 1))
+        return cr_schedule.filter(_filter_func).instructions[0][1]
+
+    def build_stretched_ecr_pulse_forward(
```

Comment:
If this is only used by the `TestRZXCalibrationBuilder` class, you can move this method to there. This helper method exists because you need to write similar test for both ECR and CX device (for now we don't have one for ECR and having the helper method seems bit redundant though). Same for `build_stretched_ecr_pulse_forward` and `build_stretched_pulse_forward`. Then these names will be slightly cleaner, i.e.

- `.build_stretched_ecr_pulse_forward` -> `TestRZXCalibrationBuilder.build_forward`
- `.build_stretched_ecr_pulse_backward` -> `TestRZXCalibrationBuilder.build_backward`
-  `.build_stretched_pulse_forward` -> `TestRZXCalibrationBuilderNoEcho.build_forward`

Reply:
Moved the helper methods to the appropriate class and simplified naming.